packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-timeline/atomic-insight-user-actions-timeline.tsx

```diff
@@ -0,0 +1,222 @@
+          onClick={this.toggleFollowingSessions.bind(this)}
+          class="btn-text-primary flex items-center p-2"
+          data-testid={
+            this.followingSessionsAreVisible
+              ? 'hide-following-sessions'
+              : 'show-following-sessions'
+          }
+        >
+          <atomic-icon
+            class="mr-1.5 h-3 w-3"
+            icon={
+              this.followingSessionsAreVisible ? ArrowDownIcon : ArrowUpIcon
+            }
+          ></atomic-icon>
+          {this.followingSessionsAreVisible
+            ? this.bindings.i18n.t('hide-following-sessions')
+            : this.bindings.i18n.t('show-following-sessions')}
+        </button>
+      </div>
+    );
+  }
+
+  private renderTogglePrecedingSessionsButton() {
+    return (
+      <div class="flex justify-center p-2">
+        <button
+          onClick={this.togglePrecedingSessions.bind(this)}
+          class="btn-text-primary flex items-center p-2"
+          data-testid={
+            this.precedingSessionsAreVisible
+              ? 'hide-preceding-sessions'
+              : 'show-preceding-sessions'
+          }
+        >
+          <atomic-icon
+            class="mr-1.5 h-3 w-3"
+            icon={
+              this.precedingSessionsAreVisible ? ArrowUpIcon : ArrowDownIcon
+            }
+          ></atomic-icon>
+          {this.precedingSessionsAreVisible
+            ? this.bindings.i18n.t('hide-preceding-sessions')
+            : this.bindings.i18n.t('show-preceding-sessions')}
+        </button>
+      </div>
+    );
+  }
+
+  private renderFollowingSessionsSection() {
+    if (!this.userActionsState.timeline?.followingSessions?.length) {
+      return null;
+    }
+    return [
+      this.renderToggleFollowingSessionsButton(),
+      <div class="separator mx-1 rounded"></div>,
+      this.followingSessionsAreVisible
+        ? this.renderSessions(
+            this.userActionsState.timeline?.followingSessions,
+            () => <div class="separator mx-1 mt-4 rounded"></div>,
+            'following-session'
+          )
+        : null,
+    ];
+  }
+
+  private renderPrecedingSessionsSection() {
+    if (!this.userActionsState.timeline?.precedingSessions?.length) {
+      return null;
+    }
+    return [
+      <div class="separator mx-1 mt-4 rounded"></div>,
+      this.precedingSessionsAreVisible
+        ? this.renderSessions(
+            this.userActionsState.timeline?.precedingSessions,
+            () => <div class="separator mx-1 mt-4 rounded"></div>,
+            'preceding-session'
+          )
+        : null,
+      this.renderTogglePrecedingSessionsButton(),
+    ];
+  }
+
+  private renderTimeline() {
+    return (
+      <div>
+        {this.renderFollowingSessionsSection()}
+        <div class="mt-4">
+          <atomic-insight-user-actions-session
+            startTimestamp={this.userActionsState?.timeline?.session?.start!}
+            userActions={this.userActionsState?.timeline?.session?.actions!}
```

Comment:
```suggestion
            startTimestamp={this.userActionsState?.timeline?.session?.start}
            userActions={this.userActionsState?.timeline?.session?.actions}
```

are those necessary here? if `renderTimeline` is called it means there are userActionsAvailable and no error right?

Reply:
I don't think so since in the preprocessing we are checking before building the timeline the following:
<img width="656" alt="image" src="https://github.com/user-attachments/assets/2e8c59c1-906a-4606-8d97-375fa5b0ad98">

Reply:
like you can see here: https://github.com/coveo/ui-kit/blob/47a1cae82a461c81876773a2d329da909e0b7e49/packages/headless/src/features/insight-user-actions/insight-user-actions-state.ts#L5

the session is optional, so it can be undefined.
For example when no actions are found from the API, session is `undefined`.